### PR TITLE
Use @shopify/typescript-configs as a base

### PR DESCRIPTION
**Pull request by BPScott**
## Description

Simplify our to tsconfig.base.json by inheriting from the pan-org ts configs.

The pan org config enables strict mode but we're not compliant yet, so opt out of the currently problematic sub-checks for now. We can come back to enabling these items in follow up PRs

### CODE REVIEW

File: config/typescript/tsconfig.base.json [hunk 1]
```diff
@@ -1,33 +1,15 @@
...
-    "declarationMap": true,
-    "pretty": true,
-    "moduleResolution": "node",
-    "target": "es5",
-    "downlevelIteration": true,
-    "esModuleInterop": true,
-    "jsx": "react",
-    "isolatedModules": true,
-    "strictNullChecks": true,
+    // Strict mode enables these strict* and noImplicit* options.
```
Comment (atesgoral) [inline config/typescript/tsconfig.base.json:5]: While this obviously seems to work fine, could we perhaps use a hack like a `"_NOTICE": "...."` to keep this as valid JSON?

Comment (BPScott) [reply config/typescript/tsconfig.base.json]: Updated!

File: config/typescript/tsconfig.base.json [hunk 2]
```diff
@@ -1,33 +1,14 @@
...
-    "declarationMap": true,
-    "pretty": true,
-    "moduleResolution": "node",
-    "target": "es5",
-    "downlevelIteration": true,
-    "esModuleInterop": true,
-    "jsx": "react",
-    "isolatedModules": true,
-    "strictNullChecks": true,
+    "__TODOCOMMENT1": "Strict mode enables these strict* and noImplicit* options.",
```
Comment (ismail-syed) [inline config/typescript/tsconfig.base.json:5]: Interesting, what are these?

Comment (BPScott) [reply config/typescript/tsconfig.base.json:5]: Comments are valid in tsconfig files, but @atesgoral asked about keeping valid json; https://github.com/Shopify/quilt/pull/1811#discussion_r605777199 


So I made up some keys that do nothing.

Personally I'm not a fan and would prefer actual comments but it's not a hill i'm gonna die on

File: config/typescript/tsconfig.base.json (hunk 1 above)
Comment (BPScott) [reply config/typescript/tsconfig.base.json]: Ah nope. TS type check says no:

```
config/typescript/tsconfig.base.json:6:5 - error TS5023: Unknown compiler option '__TODOCOMMENT2'.

6     "__TODOCOMMENT2": "Disable them temporarily while we fix up problematic behaviour from the time before we wanted strict mode",
      ~~~~~~~~~~~~~~~~
```

File: config/typescript/tsconfig.base.json (hunk 2 above)
Comment (BPScott) [reply config/typescript/tsconfig.base.json]: Ah well, turns out arbitary strings cause type check to fail. so back to real comments.

**Comment by atesgoral** [comment]
I followed the config trail to do a diligent review:

https://github.com/Shopify/web-configs/blob/main/packages/typescript-configs/library.json:

```json
{
  "extends": "./dom.json",
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "noEmit": false,
    "noEmitHelpers": true
  }
}
```

./dom.json:

```json
{
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react",
    "lib": [
      "dom",
      "dom.iterable",
      "es2017",
      "esnext.asynciterable",
      "scripthost"
    ],
    "module": "esnext",
    "target": "es2017"
  },
  "files": ["./definitions/images.d.ts", "./definitions/styles.d.ts"]
}
```

./base.json:

```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "esModuleInterop": true,
    "isolatedModules": true,
    "experimentalDecorators": true,
    "lib": ["dom", "es2018", "esnext.asynciterable"],
    "moduleResolution": "node",
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true
  }
}
```

Squashed:

```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "esModuleInterop": true,
    "isolatedModules": true,
    "experimentalDecorators": true,
    "lib": ["dom", "es2018", "esnext.asynciterable"],
    "moduleResolution": "node",
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    ?"resolveJsonModule": true,
    ?"skipLibCheck": true,
    ?"strict": true,
    "jsx": "react",
    ?"lib": [
      "dom",
      "dom.iterable",
      "es2017",
      "esnext.asynciterable",
      "scripthost"
    ],
    ?"module": "esnext",
    ?"target": "es2017",
    "declaration": true,
    "declarationMap": true,
    ?"noEmit": false,
    ?"noEmitHelpers": true
  },
  "files": ["./definitions/images.d.ts", "./definitions/styles.d.ts"]
}
```

These are the properties that are different/missing from the removed bits:

- [ ]  ?"resolveJsonModule": true,
- [ ]  ?"skipLibCheck": true,
- [ ]  ?"strict": true,
-  ?"lib": [
- [ ]    "dom",
- [ ]    "dom.iterable",
- [ ]    "es2017",
- [ ]    "esnext.asynciterable",
- [ ]    "scripthost"
-   ],
- [ ]  ?"module": "esnext",
- [ ]  ?"target": "es2017",
- [ ]  ?"noEmit": false,
- [ ]  ?"noEmitHelpers": true

Just want to ensure all of these are justified ✅

**Comment by atesgoral** [comment]
LGTM! Thanks for the detailed explanations!

**Review by atesgoral — APPROVED** [review]

**Pull request merged**